src/sql/workbench/services/commandLine/common/commandLineService.ts

```diff
@@ -108,20 +110,48 @@ export class CommandLineService implements ICommandLineProcessing {
 			}
 			await this._commandService.executeCommand(commandName, connectedContext);
 		} else if (profile) {
-			if (this._statusBarService) {
-				this._statusBarService.setStatusMessage(localize('openingNewQueryLabel', 'Opening new query:') + profile.serverName, 2500);
+			// If we were given a file and it was opened with the sql editor,
+			// we want to connect the given profile to to it.
+			// If more than one file was passed, only show the connection dialog error on one of them.
+			if (args._ && args._.length > 0) {
+				await args._.forEach((f, i) => this.processFile(URI.file(f).toString(), profile, i === 0));
 			}
-			// Default to showing new query
-			try {
-				await TaskUtilities.newQuery(profile,
-					this._connectionManagementService,
-					this._queryEditorService,
-					this._objectExplorerService,
-					this._editorService);
-			} catch (error) {
-				warn('unable to open query editor ' + error);
-				// Note: we are intentionally swallowing this error.
-				// In part this is to accommodate unit testing where we don't want to set up the query stack
+			else {
+				// Default to showing new query
+				if (this._statusBarService) {
+					this._statusBarService.setStatusMessage(localize('openingNewQueryLabel', 'Opening new query:') + profile.serverName, 2500);
+				}
+				try {
+					await TaskUtilities.newQuery(profile,
+						this._connectionManagementService,
+						this._queryEditorService,
+						this._objectExplorerService,
+						this._editorService);
+				} catch (error) {
+					warn('unable to open query editor ' + error);
+					// Note: we are intentionally swallowing this error.
+					// In part this is to accommodate unit testing where we don't want to set up the query stack
+				}
+			}
+		}
+	}
+
+	private async processFile(uriString: string, profile: IConnectionProfile, warnOnConnectFailure: boolean): Promise<void> {
```

Comment:
Doc comment here would be nice...processFile isn't exactly descriptive of what this method is doing.